manifests/init.pp

```diff
@@ -90,13 +91,20 @@
       $_package_file_name = undef
     }
 
+    # Allow for normalizing package_version for the package provider via _package_version.
+    # This only needs to be passed through to install, as elsewhere we want to
+    # use the full version string for comparisons.
+    if !$_package_version {
```

Comment:
Instead of relying on puppet's falsiness, maybe we should check `$_package_version != undef`?